File: week5/diamond.cpp
#include<iostream>
using namespace std;
class ClassA
{
public:
void aClass()
{
cout<<"ClassA\n";
}
};
class ClassB: public ClassA
{
public:
void bClass()
{
cout<<"ClassB\n";
}
};
class ClassC: public ClassA
{
public:
void cClass()
{
cout<<"ClassC\n";
}
};
class ClassD: public ClassC,public ClassB
{
public:
void dClass()
{
cout<<"ClassD\n";
}
};
int main()
{
ClassD obj1;
//obj1.aClass();
//aClass() method becomes ambiguous here
obj1.bClass();
obj1.cClass();
obj1.dClass();
return 0;
}

File: week5/diamond.java
class mainclass
{
public
void aClass()
{
System.out.println("mainclass");
}
}
class subclass1 extends mainclass
{
public
void bClass()
{
System.out.println("subclass1");
}
}
class subclass2 extends mainclass
{
public
void cClass()
{
System.out.println("subclass2");
}
}
public class diamond {
public static void main(String args[])
{
subclass1 obj1 = new subclass1();
subclass2 obj2 = new subclass2();
obj1.aClass();
obj1.bClass();
obj2.aClass();
obj2.cClass();
}
}
